src/components/BlockActionsModal.tsx

```diff
@@ -0,0 +1,67 @@
+import { Text, TouchableOpacity } from 'react-native';
+import { View } from 'react-native';
+import Modal from 'react-native-modal';
+import React from 'react';
+import Ionicon from '@expo/vector-icons/Ionicons';
+
+interface BlockActionsModalProps {
+  deleteWorkout: () => void;
+  setModalVisible: (bool: boolean) => void;
+  renameWorkout: () => void;
+  copyWorkout: () => void;
+}
+const BlockActionsModal: React.FC<BlockActionsModalProps> = ({
+  deleteWorkout,
+  setModalVisible,
+  renameWorkout,
+  copyWorkout,
+}) => {
+  return (
+    <Modal
+      style={{ margin: 0 }}
+      isVisible={true}
+      onBackdropPress={() => setModalVisible(false)}
+      onSwipeComplete={() => setModalVisible(false)}
+      swipeDirection="down"
+    >
+      <View className="absolute top-3/4 h-full w-full rounded-t-3xl border-2 bg-white ">
+        <View className="my-3 h-1 w-16 self-center rounded-lg bg-gray-400"></View>
+        <TouchableOpacity
+          accessibilityRole="button"
+          className="justify-left ml-9 flex flex-row items-center space-x-4 p-1"
+          hitSlop={{ top: 20, bottom: 20, right: 20 }}
```

Comment:
The hit slops overlap with each other, resulting in accidentally clicking one button when trying to hit another. I'd just get rid of the hit slops and give the buttons more space between each other. Also, if we could add the light grey borders like in the figma you made I think that'd look really nice and help separate the buttons visually.

Reply:
If you space them out with margin, add a light rounded border, and more padding inside, that will make them easier to hit.